core/src/test/java/org/apache/iceberg/util/TestJsonUtil.java

```diff
@@ -195,6 +195,26 @@ public void getBool() throws JsonProcessingException {
     assertThat(JsonUtil.getBool("x", JsonUtil.mapper().readTree("{\"x\": false}"))).isFalse();
   }
 
+  @Test
+  public void getBoolOrNull() throws JsonProcessingException {
+    assertThat(JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{}"))).isNull();
+
+    assertThat(JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{\"x\": null}"))).isNull();
+
+    assertThatThrownBy(
+            () -> JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{\"x\": \"23\"}")))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse to a boolean value: x: \"23\"");
+
+    assertThatThrownBy(
+            () -> JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{\"x\": \"true\"}")))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse to a boolean value: x: \"true\"");
+
+    assertThat(JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{\"x\": true}"))).isTrue();
+    assertThat(JsonUtil.getBoolOrNull("x", JsonUtil.mapper().readTree("{\"x\": false}"))).isFalse();
```

Comment:
Super nit, feel free to ignore but I'd find this test a bit easier to read if we just move all the happy cases together and then the failure cases after